x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/file/FileUserPasswdStore.java

```diff
@@ -84,14 +82,14 @@ public AuthenticationResult verifyPassword(String username, SecureString passwor
         if (hash == null) {
             return AuthenticationResult.notHandled();
         }
-        if (hasher.verify(password, hash) == false) {
+        if (Hasher.verifyHash(password, hash) == false) {
             return AuthenticationResult.unsuccessful("Password authentication failed for " + username, null);
         }
         return AuthenticationResult.success(user.get());
     }
 
     public boolean userExists(String username) {
-        return users.containsKey(username);
+        return users != null && users.containsKey(username);
```

Comment:
why the null check? I specifically changed this a little while ago

Reply:
I have no specific recollection, I assume this was leftover from an earlier approach/change that was discarded. Will revert.